src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddleware.cs

```diff
@@ -127,13 +128,22 @@ public async Task Invoke(HttpContext context)
         // Assumes the response headers have not been sent.  If they have, still attempt to write to the body.
         private Task DisplayException(ErrorContext errorContext)
         {
-            var compilationException = errorContext.Exception as ICompilationException;
-            if (compilationException != null)
+            var httpContext = errorContext.HttpContext;
+            var acceptHeader = httpContext.Request.Headers[HeaderNames.Accept].ToString();
+
+            // If the client does not ask for HTML just format the exception as plain text
+            if (acceptHeader != null && !acceptHeader.Contains("text/html", StringComparison.OrdinalIgnoreCase))
+            {
+                httpContext.Response.ContentType = "text/plain";
+                return httpContext.Response.WriteAsync(errorContext.Exception.ToString());
+            }
+
+            if (errorContext.Exception is ICompilationException compilationException)
             {
-                return DisplayCompilationException(errorContext.HttpContext, compilationException);
+                return DisplayCompilationException(httpContext, compilationException);
```

Comment:
Is the reason not to go down this code path that it's pretty hardcoded to HTML?

OR that the scenario doesn't match? Right now compilation exception => views => browser...

Reply:
I didn't think it warranted special formatting (especially since runtime compilation is off by default). Do you think we should format compilation errors in plaintext differently? Bare in mind you’d only hit this if you were using test server or postman making a request for a view that wasn’t already compiled *and* there happens to be an error.

Reply:
I missed the previous conversation on this. Fine, I can see why this isn't absolutely necessary.

Reply:
Yes we discussed it offline. If it becomes useful in the future then we can do so.